src/invoice.rs

```diff
@@ -0,0 +1,416 @@
+//! Contains the main invoice object definition and its implementation. This implementation includes
+//! signing and helper functions for accessing the bindle name
+
+use ed25519_dalek::{Keypair, PublicKey, Signature as EdSignature, Signer};
+use semver::{Compat, Version, VersionReq};
+use serde::{Deserialize, Serialize};
+
+use std::convert::TryInto;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use super::{AnnotationMap, BindleSpec, Group, Parcel, Signature, SignatureError, SignatureRole};
+
+/// The main structure for a Bindle invoice.
+///
+/// The invoice describes a specific version of a bindle. For example, the bindle
+/// `foo/bar/1.0.0` would be represented as an Invoice with the `BindleSpec` name
+/// set to `foo/bar` and version set to `1.0.0`.
+///
+/// Most fields on this struct are singular to best represent the specification. There,
+/// fields like `group` and `parcel` are singular due to the conventions of TOML.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct Invoice {
+    pub bindle_version: String,
+    pub yanked: Option<bool>,
+    pub bindle: BindleSpec,
```

Comment:
This feels strange. The top-level object for the library is Invoice. Now it's in its own module... which is fine... except that all of its children (BindleSpec, Group, etc.) are now in a _parent_ module. That feels inverted to have the children objects defined in the parent module.